### PR TITLE
Wrap envs with multidiscrete action space into discrete action space

### PR DESCRIPTION
**Pull request by seungjaeryanlee**
Some Gym environments have multidiscrete action spaces. (ex. [Unity's Obstacle Tower environment](https://github.com/Unity-Technologies/obstacle-tower-env)) This PR implements `MultiDiscreteToDiscreteWrapper`, an environment wrapper that wraps such environments to make them into an environment with discrete action spaces.

Resolves Issue #97

### CODE REVIEW

File: tf_agents/environments/wrappers.py [hunk 1]
```diff
@@ -719,3 +719,45 @@ def _step(self, action):
...
+      shape=(np.prod(env.action_spec().shape),),
+      dtype=env.action_spec().dtype,
+      minimum=0,
+      maximum=np.prod(env.action_spec().shape),
+    )
+
+  def _discrete_to_multidiscrete(self, discrete_action):
+    """Convert discrete action to multidiscrete action from original env.
+
+    The conversion follows lexicographical order.
```
Comment (oars) [inline tf_agents/environments/wrappers.py:747]: Why lexicographical order? I think the mapping may be clearer if it simply iterated over the multiple discrete spaces.

File: tf_agents/environments/wrappers.py [hunk 2]
```diff
@@ -719,3 +719,45 @@ def _step(self, action):
 
     time_step = self._env.step(action)
     return self._add_history(time_step, action)
+
+
+@gin.configurable
+class MultiDiscreteToDiscreteWrapper(PyEnvironmentBaseWrapper):
+  """Converts an environment with Multidiscrete action space into a Discrete one."""
```
Comment (oars) [inline tf_agents/environments/wrappers.py:726]: Document how the multiple discrete spaces are mapped to a single one. You can show a remapping of the sample action in the multidiscrete space from gym:

https://github.com/openai/gym/blob/master/gym/spaces/multi_discrete.py

File: tf_agents/environments/wrappers_test.py [hunk 3]
```diff
@@ -914,5 +914,60 @@ def test_observation_and_action_stacked(self):
...
+  def test_new_and_old_actions_have_one_to_one_correspondence(self):
+    env = CustomMultiDiscretePyEnv()
+    wrapped_env = wrappers.MultiDiscreteToDiscreteWrapper(CustomMultiDiscretePyEnv())
+    ts = env.step((0, 0))
+    wrapped_ts = wrapped_env.step(0)
+    self.assertEqual(ts.observation.tolist(), wrapped_ts.observation.tolist())
+
+    env = CustomMultiDiscretePyEnv()
+    wrapped_env = wrappers.MultiDiscreteToDiscreteWrapper(CustomMultiDiscretePyEnv())
+    ts = env.step((0, 1))
```
Comment (oars) [inline tf_agents/environments/wrappers_test.py:955]: Would this fail if you did wrappedenv.step(2) Make sure the test covers a case where the ordering of the mapping changes.

Maybe make the cusotmenvs position change by 10*action[0] + action[1]?

**Comment by tfboyd** [comment]
Closing, we worked hard to catch up on some of the really old PRs, but this one may have to be a casualty. We have gotten much better at working the PRs faster.

**Pull request closed without merging**